cpu/esp_common/lib_printf.c

```diff
@@ -46,6 +46,16 @@ static int _lib_printf(int level, const char* tag, const char* format, va_list a
 
     int len = vsnprintf(_printf_buf, PRINTF_BUFSIZ - 1, format, arg);
 
+    if (len < 0) {
+        ESP_EARLY_LOGI(tag, "Failed to format print");
+        return 0;
+    }
+
+    /* Did the output got truncated? */
+    if ((unsigned) len > PRINTF_BUFSIZ) {
+        len = PRINTF_BUFSIZ;
```

Comment:
I think this needs to be `PRINTF_BUFSIZ-1` in both cases 

also the return value of this function does not get recalculated when the number of printed chars is reduced in the for loop (this might be expected by the esp SDK)

Reply:
I force pushed. Please check again.